client/src/components/Carousel/Carousel.js

```diff
@@ -0,0 +1,51 @@
+import { useState, useEffect } from "react";
+import { ReactComponent as Left } from "assets/left.svg";
+import { ReactComponent as Right } from "assets/right.svg";
+import "./Carousel.css";
+
+const Carousel = ({ images, loading }) => {
+  const [currentImage, setCurrentImage] = useState(0);
+
+  useEffect(() => {
+    setCurrentImage(0);
+  }, [loading]);
+
+  const previousImage = () => {
+    setCurrentImage((currentImage - 1 + images.length) % images.length);
+  };
+
+  const nextImage = () => {
+    setCurrentImage((currentImage + 1) % images.length);
+  };
+
+  return (
+    <div className="carousel-container">
+      {images.length === 0 ? (
+        <p className="no-photos" data-testid="no-photos">
+          no photos
+        </p>
+      ) : (
+        <>
+          <span
```

Comment:
wondering why you chose to use a span here for the arrow? Can you see any downsides of using a span for an interactive element?

Reply:
Thinking about it now it probably wasn't a good idea to use span. I remember using it similarly in the past but it was for non-interactive icons. The downsides of using it for an interactive element are that it doesn't have the default styling for interactive elements so those have to be manually added, it also doesn't have the necessary attributes to make it accessible, and it doesn't have any semantic meaning so search engines won't recognize the purpose of the element.